src/openfermion/ops/_code_operator.py

```diff
@@ -0,0 +1,298 @@
+import copy
+
+import numpy
+import scipy
+import scipy.sparse
+
+from openfermion.ops import SymbolicBinary
+
+
+def _shift_decoder(decode, shift_constant):
+    """ Shifts the indices of a decoder by a constant.
+    
+    Args:
+        decode (list of SymbolicBinary): a decoder
+        shift_constant (int): the qubit index that corresponds to the offset.
+
+    Returns (list of SymbolicBinary):  shifted decoder
+    """
+    decode_shifted = []
+    for entry in decode:
+        tmp_entry = copy.deepcopy(entry)
+        tmp_entry._shift(shift_constant)
+        decode_shifted.append(tmp_entry)
+    return numpy.array(decode_shifted)
```

Comment:
This doesn't match the doc strings, which say a list is returned. I am surprised this works - I did not know you could have a numpy array populated with a custom class as entries.